pkg/pcap/pcaps.go

```diff
@@ -0,0 +1,156 @@
+package pcap
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//
+// This is a big Pcaps struct holding caches for different types of pcap files
+// to be managed (pcap files per process, per containers and per commands). It
+// would be hard to keep all possible pcap files open forever (as tracee might
+// trace tons of processes, containers or commands).
+//
+// That is what the pcapCache interface is for. It can be implemented by any
+// different way of keeping pcap files and maintain those pcap files cached
+// while they're being used.
+//
+// At the end we have the pcap interface itself. It describes a pcap file being
+// kept opened on behalf of a process, a container or a command.
+//
+// NOTE: Pcaps is not thread safe, should be called from a single routine.
+// TODO: might replace cache_XXX and pcap_XXX files with generics.
+//
+
+//
+// All Pcap Implementations at Once
+//
+
+type Pcaps struct {
+	processes  pcapCache // ProcessPcap Cache (indexed by host_tid)
```

Comment:
We can possibly reduce this to one field of `[]pcapCache` then all operations simply range on the caches.
Later if we want to add another cache type it would only require adding a new config flag and appending a cache (of course if you think future caches won't be added this doesn't really matter).

Reply:
Yep, could be. let's see what generics will look like (if they don't compromise readability too much) and then check if its worth it or keep it as is for now. Will get back to this in the next round.